benchmarks/src/main/kotlin/benchmarks/common/ChannelProdConsBenchmarkIteration.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2016-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package benchmarks.common
+
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.*
+import kotlinx.coroutines.selects.*
+import java.util.concurrent.*
+
+/**
+ * Runs a batch send-receive pairs under the specified workload.
+ */
+public abstract class ChannelProdConsBenchmarkIteration(
```

Comment:
Currently, the only possible way to understand the structure of the benchmark (what does it measure, how does it split the workload, what's the relationship between producers and consumers, what caveats does it have etc.) can be understood only from studying the source code, which doesn't seem really maintainable.

Please describe the top-level process in the root KDoc